api/v1alpha7/openstackcluster_conversion.go

```diff
@@ -348,14 +348,11 @@ func restorev1alpha7ClusterStatus(previous *OpenStackClusterStatus, dst *OpenSta
 }
 
 func restorev1beta1ClusterStatus(previous *infrav1.OpenStackClusterStatus, dst *infrav1.OpenStackClusterStatus) {
-	// ReferencedResources have no equivalent in v1alpha7
-	if previous.Bastion != nil {
-		dst.Bastion.Resolved = previous.Bastion.Resolved
+	if previous == nil || dst == nil {
```

Comment:
this check also happens in `restorev1beta1BastionStatus`, can it be removed?

Reply:
In `restorev1beta1BastionStatus` it's BastionStatus, not ClusterStatus. I think we need both.

Reply:
Here we are checking `previous` and `dst`. Then we pass on `previous.Bastion` and `dst.Bastion`, so I would say both checks are needed.